pkg/serviceprovider/serviceprovider.go

```diff
@@ -76,6 +76,15 @@ type ServiceProvider interface {
 
 	// Validate checks that the provided object (token or binding) is valid in this service provider
 	Validate(ctx context.Context, validated Validated) (ValidationResult, error)
+
+	//RefreshToken(ctx context.Context, token *api.Token, clientId string, clientSecret string) (*api.Token, error)
+}
+
+// RefreshableTokenServiceProvider represents a service provider which issues refreshable access tokens
+type RefreshableTokenServiceProvider interface {
```

Comment:
Sorry i'm posting it too late.. haven't you considered making that optional operation by using capability ? (same as we did for file downloads)? Just to use same techinque.

Reply:
I am able to refactor it this way if you would like, but I honestly don't like it. Right now, It would mean adding a method that returns `nil` to every provider except GitLab and bloat the `serviceProvider` interface + creating `refreshTokenCapability` interface and struct, which would have to be initialized together with the gitlab struct. I think how it is right now is cleaner.

Reply:
I don't think it's a bad idea to do this with another `capability`. At least it would be unified what service provider can do. We can't do everything with interfaces, because for example OAuthCapability depends on the configuration (decided at runtime).

Reply:
Done.

Reply:
How valuable the information about potential ability to provide token refresh on SP level is?
For example (AFAIK) GitHub provides several types of tokens with or without refresh, like:  
- OAuth token with "classic" OAuth flow - not refreshable
- Classic PAT - not refreshable
- Fine grained PAT - not refreshable
- Token provided with GitHub App flow - refreshable
So, it has refreshTokenCapability in general, or it does not?